src/Voron/Impl/Journal/WriteAheadJournal.cs

```diff
@@ -1000,21 +1017,17 @@ public bool WaitForTaskToBeDone(Action task)
                         return false;
                     }
                 }
-
+                
                 public void RunTaskIfNotAlreadyRan()
                 {
-                    Debug.Assert(Monitor.IsEntered(_lock));
-
-                    if (_token.IsCancellationRequested)
```

Comment:
Why we removed that?

Reply:
I passed it to [_token.ThrowIfCancellationRequested();](https://github.com/ravendb/ravendb/pull/8276/files/cd455fc9b8f00db11d6453bba4822982afda6e07#diff-33af42ec0439152b32cc47c6739f3211R1029). 
For now it will throw exception the `LockTaskResponsible` will save it and [throw it again](https://github.com/ravendb/ravendb/pull/8276/files/cd455fc9b8f00db11d6453bba4822982afda6e07#diff-33af42ec0439152b32cc47c6739f3211R1010) in `WaitForTaskToBeDone` like all exceptions
and just in case of `OperationCanceledException` it will [catch it and return false](https://github.com/ravendb/ravendb/pull/8276/files/cd455fc9b8f00db11d6453bba4822982afda6e07#diff-33af42ec0439152b32cc47c6739f3211R1015)